docs/guides/integrating-cert-manager.md

```diff
@@ -0,0 +1,290 @@
+4.	A DNS resolvable domain name is required. It must resolve to the public endpoint of the NKG deployment, and this
+    public endpoint must be an external IP address or alias accessible over the internet. The process here will depend
+    on your DNS provider. This DNS name will need to be resolvable from the Let’s Encrypt servers, which may require
+    that you wait for the record to propagate before it will work.
+
+## Overview
+
+![cert-manager ACME Challenge and certificate managment with Gateway API](/docs/images/cert-manager-gateway-workflow.png)
+
+At a high level, the process looks like this:
+
+1. We deploy cert-manager and create a ClusterIssuer which specifies Let’s Encrypt as our CA and Gateway as our ACME
+   HTTP01 Challenge solver
+2. We create a Gateway resource for our domain (cafe.example.com) and configure cert-manager integration using an
+   annotation.
+3. This kicks off the certificate issuance process – cert-manager contacts Let’s Encrypt to obtain a certificate, Let’s
+   Encrypt starts the ACME challenge, and once the domain has been verified, the certificate is issued. Cert-manager
+   stores this certificate and the signed keypair as Kubernetes resources. The keypair is stored in a Kubernetes secret
+   that is referenced by the Gateway resource.
+4. We deploy our application and our HTTPRoute which defines our routing rules. The routing rules defined configure
+   NGINX to direct requests to https://www.cafe.example.com/coffee to our coffee-app application, and to use the https
+   listener defined in our Gateway resource.
+5. NGINX is configured to terminate HTTPS traffic from clients using the signed keypair contained in the Kubernetes
+   secret referenced in the Gateway resource.
+6. When the client connects to https://www.cafe.example.com/coffee, the request is routed to the coffee-app application
+   and the communication is secured using the signed keypair contained in the cafe-secret Secret.
+7. The certificate will be automatically renewed when it is close to expiry, the Secret will be updated using the new
+   Certificate, and NKG will dynamically update the keypair on the filesystem used by NGINX for HTTPS termination once
+   the Secret is updated.
+
+## Details
+
+### Step 1 – Deploy cert-manager
+
+The first step is to deploy cert-manager onto the cluster.
+
+1. Add the Helm repository
+
+```shell
+    helm repo add jetstack https://charts.jetstack.io
+	helm repo update
+```
+
+1. Install cert-manager, and enable the GatewayAPI feature gate:
+
+```shell
+helm install \
+  cert-manager jetstack/cert-manager \
+  --namespace cert-manager \
+  --create-namespace \
+  --version v1.12.0 \
+  --set installCRDs=true \
+  --set "extraArgs={--feature-gates=ExperimentalGatewayAPISupport=true}"
+```
+
+### Step 2 – Create a ClusterIssuer
+
+Next we need to create a ClusterIssuer. Issuers, and ClusterIssuers, are Kubernetes resources that represent certificate
```

Comment:
I don't think we need to introduce Issuer resource here, so that we have less new information for the reader